src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -59,8 +65,42 @@ internal struct NameDeclarationInfo
                 return default(NameDeclarationInfo);
             }
 
+            private static bool IsPossibleOutVariableDeclaration(SyntaxToken token, SemanticModel semanticModel, int position,
+                ITypeInferenceService typeInferenceService, CancellationToken cancellationToken, out NameDeclarationInfo result)
+            {
+                if (!token.IsKind(SyntaxKind.IdentifierToken) || !(token.Parent.IsKind(SyntaxKind.IdentifierName)))
+                {
+                    result = default;
+                    return false;
+                }
+
+                var argument = token.Parent.Parent as ArgumentSyntax // Foo(out var $$
+                    ?? token.Parent.Parent.Parent as ArgumentSyntax; // Foo(out var a$$
```

Comment:
It's actually the same token either way (the token after `out`, eg `var`). However, once you start typing the name (so like `out var foo`), we start parsing that as a DeclarationExpressionSyntax. Without the name, `var` is a direct child of the ArgumentSyntax.